File: project_wecc/WECC_dataset/network_bus_clean.jl
using CSV
using DataFrames

cd(@__DIR__)
data_bus = CSV.read("WECC_Bus_all.csv")

# clear unit name
num_bus = size(data_bus)[1]
for l in 1:num_bus
    name_str = split(data_bus[l, :BusName], "-")   # split bus name by - if exist
    if length(name_str) > 1
        data_bus["BusName"][l] = name_str[2]
    end
end

CSV.write("WECC_Bus_all_julia.csv", data_bus)

File: project_wecc/WECC_dataset/network_bus_clean.py
import pandas as pd
import numpy as np
import csv

data_bus = pd.read_csv("WECC_Bus_all.csv", encoding="ISO-8859-1")

num_bus = data_bus.shape[0]
for l in range(num_bus):
    print(l)
    # clear unit name
    data_bus["BusName"][l] = data_bus["BusName"][l].split("-")[1]
    data_bus["BusNumber"][l] = data_bus["BusNumber"][l] - 1

data_bus.to_csv("WECC_Bus_all.csv", encoding="ISO-8859-1")
